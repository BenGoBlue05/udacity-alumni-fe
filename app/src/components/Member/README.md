## Member Component
A component that ...

### Example

```js
<Member />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
