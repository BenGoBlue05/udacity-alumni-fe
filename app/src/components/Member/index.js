import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Heading from 'grommet-udacity/components/Heading';
import Box from 'grommet-udacity/components/Heading';
import Paragraph from 'grommet-udacity/components/Paragraph';
import { Avatar } from 'components';

const Member = ({
  user,
}) => (
  <Box
    className={styles.member}
    align="center"
    justify="center"
    size="large"
  >
    <Avatar src={user.avatar} />
    <Heading tag="h3" align="center">
      {user.name}
    </Heading>
    <Paragraph>
      {`${user.bio.slice(0, 120)}...`}
    </Paragraph>
  </Box>
);

Member.propTypes = {
  user: PropTypes.object.isRequired,
};

export default cssModules(Member, styles);
