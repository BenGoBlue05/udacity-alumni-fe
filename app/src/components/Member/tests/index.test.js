import Member from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<Member />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Member />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
