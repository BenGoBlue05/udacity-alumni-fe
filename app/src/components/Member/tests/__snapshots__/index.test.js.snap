exports[`<Member /> should render with default props 1`] = `
<Heading
  align="center"
  justify="center"
  size="large"
  tag="h1">
  <Avatar
    src="http://overhere" />
  <Heading
    align="center"
    tag="h3" />
  <Paragraph>
    I have done this and that....
  </Paragraph>
</Heading>
`;
